modyn/supervisor/internal/triggers/datadrifttrigger.py

```diff
@@ -0,0 +1,260 @@
+        assert self.dataloader_info is not None
+        assert self.drift_dir is not None
+
+        # Run Evidently detection
+        column_mapping = ColumnMapping(embeddings={self.evidently_column_mapping_name: reference_embeddings_df.columns})
+
+        # https://docs.evidentlyai.com/user-guide/customization/embeddings-drift-parameters
+        report = Report(metrics=self.metrics)
+        report.run(
+            reference_data=reference_embeddings_df, current_data=current_embeddings_df, column_mapping=column_mapping
+        )
+        result = report.as_dict()
+        result_print = [
+            (x["result"]["drift_score"], x["result"]["method_name"], x["result"]["drift_detected"])
+            for x in result["metrics"]
+        ]
+        logger.info(
+            f"[DataDriftDetector][Prev Trigger {self.previous_trigger_id}][Dataset {self.dataloader_info.dataset_id}]"
+            + f"[Result] {result_print}"
+        )
+
+        with open(self.drift_dir / f"drift_{self.previous_trigger_id}.json", "w", encoding="utf-8") as f:
+            json.dump(result, f)
+
+        return result["metrics"][0]["result"]["drift_detected"]
+
+    def detect_drift(self, idx_start: int, idx_end: int) -> bool:
+        assert self.previous_trigger_id is not None
+        assert self.previous_data_points is not None and self.previous_data_points > 0
+        assert self.previous_model_id is not None
+        assert self.dataloader_info is not None
+        assert self.model_downloader is not None
+
+        reference_dataloader = prepare_trigger_dataloader_by_trigger(
+            self.previous_trigger_id,
+            self.dataloader_info,
+            data_points_in_trigger=self.previous_data_points,
+            sample_size=self.sample_size,
+        )
+
+        current_keys, _, _ = zip(*self.data_cache[idx_start:idx_end])  # type: ignore
+        current_dataloader = prepare_trigger_dataloader_fixed_keys(
+            self.previous_trigger_id + 1,
+            self.dataloader_info,
+            current_keys,
+            sample_size=self.sample_size,
+        )
+
+        # Download previous model as embedding encoder
+        # TODO(366) Support custom model as embedding encoder
+        if self.model_updated:
+            self.model_downloader.download(self.previous_model_id)
+            self.model_updated = False
+
+        # Compute embeddings
+        reference_embeddings_df = get_embeddings_evidently_format(self.model_downloader, reference_dataloader)
+        current_embeddings_df = get_embeddings_evidently_format(self.model_downloader, current_dataloader)
+
+        drift_detected = self.run_detection(reference_embeddings_df, current_embeddings_df)
+
+        return drift_detected
+
+    def inform(self, new_data: list[tuple[int, int, int]]) -> Generator[int, None, None]:
+        """
+        Use Generator here because this data drift trigger
```

Comment:
I think this needs a longer docstring, sorry. Could you provide some longer documentation here, especially on the sliding window approach and what is happening? Right now, the docstring mostly concerns the return type

Reply:
done.